crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -208,69 +234,34 @@ impl SystemExecutor for MultiThreadedExecutor {
             .map(|e| e.0.clone());
         let thread_executor = thread_executor.as_deref();
 
-        let SyncUnsafeSchedule {
-            systems,
-            mut conditions,
-        } = SyncUnsafeSchedule::new(schedule);
+        let environment = &Environment::new(self, schedule, world);
```

Comment:
```suggestion
        let environment = Environment::new(self, schedule, world);
```
Is this borrow needed?

Reply:
`Context` needs a borrow because `Environment` owns the `Mutex<Conditions>`, and so that we only copy one pointer into each task instead of the whole environment.  And doing the borrow here avoids having to spell out `environment: &environment` when constructing the `Context`.